TowerForge/TowerForge/Nodes/StatePopupNode.swift

```diff
@@ -8,7 +8,8 @@
 import Foundation
 import UIKit
 
-protocol StatePopupDelegate {
+protocol StatePopupDelegate: AnyObject {
+    var isPaused: Bool { get }
```

Comment:
should this logic be abstracted out instead? I am not too sure if it is neater

Reply:
My thought is that the state might not be restricted to paused and not paused, it can be other valid states like game over in which we need to stop the game too? I am not too sure just my thinking